docs/script/generatePropsTables.mjs

```diff
@@ -0,0 +1,157 @@
+      category === 'TypeLiteral' ? 'BaseComponentProps' : category;
+    return {
+      ...acc,
+      ...(isCategoryToUse && {
+        [categoryKey]: {
+          ...acc[categoryKey],
+          [key]: val,
+        },
+      }),
+    };
+  }, {});
+
+const createPropsTable = (props) => {
+  return {
+    headers: ['Name', 'Type', 'Default', 'Description'],
+    rows: props
+      ? Object.values(props).map(
+          ({ name, type, defaultValue, description }) => ({
+            Name: name.replaceAll(/[{}]/g, '\\$&'),
+            Type: type.name.replaceAll(/[{}]/g, '\\$&'),
+            Default: defaultValue ? defaultValue.value.toString() : '/',
+            Description: description,
+          })
+        )
+      : [{ Name: '/', Type: '/', Default: '/', Description: '/' }],
+  };
+};
+
+const createPropsTableExpander = (data, wantedCategories) =>
+  wantedCategories.map((category) => ({
+    ExpanderItem: {
+      title: category,
+      value: category,
+      children: [
+        {
+          table: createPropsTable(data[category]),
+        },
+      ],
+    },
+  }));
+
+json2md.converters.ExpanderItem = ({ title, value, children }, json2md) => `
+<ExpanderItem title="${title}" value="${value}">
+    <ReactMarkdown
+      remarkPlugins={[remarkGfm]}
+      children={\`
+${json2md(children)}
+      \`}
+    />
+  </ExpanderItem>
+`;
+
+json2md.converters.Expander = ({ ExpanderItems, displayName }, json2md) => `
+<Expander type="multiple" defaultValue={['${displayName}Props']}>
+  ${json2md([...ExpanderItems.map((item) => ({ ...item }))])}
+</Expander>
+`;
+
+json2md.converters.plainText = (text, json2md) => text;
+
+const getOutput = (displayName, propsTables) =>
+  json2md([
+    {
+      plainText: `
+import { Expander, ExpanderItem } from '@aws-amplify/ui-react';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+`,
+    },
+    { h2: 'Props' },
+    {
+      Expander: {
+        ExpanderItems: propsTables,
+        displayName,
+      },
+    },
+  ]);
+
+for await (const filePath of globbyStream(
+  path.join(__dirname, '../../packages/react/src/primitives/**/index.ts')
+)) {
+  const { props, displayName } = getData(filePath);
+
+  const targetPath = path.resolve(
+    __dirname,
+    `../src/pages/components/${displayName?.toLowerCase()}`
+  );
+
+  if (!existsSync(targetPath)) continue;
+  if (!props || !displayName) continue;
+
+  const wantedCategories = getWantedCategories(displayName);
+  const categorizedProps = categorizeProps(props, wantedCategories);
+  const propsTables = createPropsTableExpander(
+    categorizedProps,
+    wantedCategories
+  );
+  const output = getOutput(displayName, propsTables);
+
+  writeFileSync(
+    path.resolve(
+      targetPath,
+      `./react-auto-prop-table-${displayName.toLowerCase()}.mdx`
+    ),
+    output
+  );
+  console.log(`✅ ${displayName} Props Tables are updated.`);
+}
+
+console.log('🎉 Props Tables are all updated.');
```

Comment:
I like the console output to notify the dev what is happening! Are there error checking states where we should output something to the console if some error states?

Reply:
I thought about it, but don't know where should I check the error...it's perfect code, shouldn't be any errors 😜

Reply:
If there's any error, we would either get the error message from nodejs, or don't see the expected MDX get generated...So I didn't find where need to add any error logs. Any thoughts?

Reply:
Done